unit/src/main/cpp/subsystems/uploader.cpp

```diff
@@ -0,0 +1,68 @@
+#include <config/constants.h>
+#include <utils/utils.h>
+#include <subsystems/uploader.h>
+#include "restclient-cpp/restclient.h"
+
+//Update values using:
+//state->monitorMeasurements[constants::general::Measurements::ATMOSPHERIC_TEMP] = 1;
+//Add a new value by doing:
+//1. Update kNumMeasurements in constants::general
+//2. Add the new measurement in constants::general::Measurements
+//3. Add the string name (the json key that is used) to kMeasurementNames in upload.h
+
+Uploader::Uploader(State *state, Commands *commands, Outputs *outputs) {
+    state->lastTimeS = Utils::getUnixTimestamp();
+}
+
+void Uploader::read(State *state) {
+    //Timer
+    long now = Utils::getUnixTimestamp();
+    if (now - state->lastTimeS > constants::uploader::kUploadInterval) {
+        state->lastTimeS = now;
+        state->doUpload = true;
+    } else {
+        state->doUpload = false;
+    }
+}
+
+void Uploader::calculate(State *state, Commands *commands, Outputs *outputs) {
+    outputs->jsonString = "";
+
+    //Generate JSON String
+    if (state->doUpload) {
+        outputs->jsonString = "{\"monitor_id\": \"" + constants::uploader::kMonitorId + "\",";
+        int valuesUploaded = 0;
+        for (int i = 0; i < constants::general::kNumMeasurements; i++) {
```

Comment:
values uploaded looks like a scuffed unneeded variable.

Reply:
It's to prevent uploading when nothing is meant to be uploaded